### ISSUE TITLE
[1.12.2] Direct Connecting through Custom Main Menu forces "offline" waypoints.

### ISSUE THREAD

**Issue by Nox13last** [TeamJM/journeymap#149]
#### JourneyMap Version:

`Example: 1.12.2-5.5.4`

#### Forge Version:

`Example: 1.12.2-14.23.5.2838`

#### Describe the problem in repeatable steps:

Prerequisite: Have connected to the below-mentioned server "normally" (through the Server Browser GUI).
Prerequisite: Install [Custom Main Menu](https://www.curseforge.com/minecraft/mc-mods/custom-main-menu/)
1. Create a Main Menu button to direct-connect to a specific server, skipping the Server Browser GUI. Adding the below code to the 'Buttons' section of the _configs\CustomMainMenu\mainmenu.json_ file will do this (modifying the appropriate parameter to specify a server).
```        "directmultiplayer":
        {
            "text" : "Connect to my server",
            "posX" : -170,
            "posY" : 48,
            "width" : 175,
            "height" : 20,
            "action" : 
            {
                "type" : "connectToServer",
                "ip" : "(IP):(PORT)"
            }
        }, 
```
2. Boot up the game and press the new button.
3. Instead of using waypoints from the Journeymap's folder named after the server, it uses the "offline" folder, creating it if it does not exist.
4. To restore waypoints, return to Main Menu, use the Server Browser GUI, and log in through that (Direct Connect, Favorites, or discovered LAN game) instead.

**Comment by BrainStone** [comment]
Have you tried it with the latest version?

**Comment by Nox13last** [comment]
> 
> 
> Have you tried it with the latest version (of both JM and CMM)?

Yeah. I just updated Journeymap to 1.12.2-5.5.5 (from 5.5.4, when the ticket was created). And I downloaded CMM fresh prior to making the ticket. No difference.
I doubt this is a serious issue, given the prerequisite.

I've managed to make a batch-file workaround which seems to work, by using mklink junctions to force the server folder and the "offline" folder to access the same thing.
`mkdir journeymap
mkdir journeymap\data
mkdir journeymap\data\mp\
mkdir journeymap\data\mp\Server\
mklink /j "journeymap\data\mp\offline" "journeymap\data\mp\Server"`

**Comment by mysticdrew** [comment]
How is journeymap supposed to know what data to use when you don't have a servername?

**Comment by Nox13last** [comment]
> 
> 
> How is journeymap supposed to know what data to use when you don't have a servername?

I fail to see the relevance of this question.

For one thing, the server I used for testing prior to writing this ticket _does_ have a name. When using the Server UI to connect to it, **this is reflected correctly in normal use of the mod.** How is it even possible to run a server that doesn't _have_ a name? Don't they use a default "Minecraft Server" name if a name isn't assigned?
When bypassing the Server UI, Journeymap seems to be unable to find this information, and it uses an odd `/mp/offline` folder instead.

My guess (from an outsider's perspective) is that it uses the Server UI to download the server's details. No Server UI (connecting without going through that screen), no name, bad folder. And yet, using the Direct IP Connect button would still scrape this information properly.

**TL;DR:** _The server **does** have a name. JM doesn't find it when using a mod to bypass the Server Browser._

**Comment by mysticdrew** [comment]
The server does have a name. JM doesn't find it when using a mod to bypass the Server Browser.
Exactly... if there is not a name associated with it, we have no way to know what server it is and what data to use. 

This is not a bug

**Comment by mysticdrew** [comment]
When you direct connect "Minecraft" not Journeymap assigns the name to the server of "Minecraft Server" so if you name your server "Minecraft Server" it will use the same data no matter if you direct connect or click the connect button on that specific server. 

If you use CMM's UI to connect, minecraft has no idea what the server name is. Journeymap saves serverdata based on the server's name that we retrieve through minecraft. 

This is a bug for CMM, tell them you want to assign a name to the net.minecraft.client.multiplayer.ServerData.serverName field when you create custom button.

**Comment by mysticdrew** [comment]
There is already an issue regarding this on their issue tracker. 
https://github.com/lumien231/Custom-Main-Menu/issues/372

**Comment by mysticdrew** [comment]
I created a pull request to fix it in CMM. 
https://github.com/lumien231/Custom-Main-Menu/pull/393
